src/relations.d.ts

```diff
@@ -14,3 +14,8 @@ export type RelationTypeFunction = (attribute: string) => string;
  * The relationTypes option can be a function or an object
  */
 export type RelationTypeOpt = RelationTypeMap | RelationTypeFunction;
+
+export interface RelationOpts {
```

Comment:
What about moving this `boolean | ...` to the `src/relations.d.ts` file

Maybe like:
```
export type RelationOpt = boolean | RelationDetails
interface RelationDetails {
  included?: boolean,
  fields?: string[]
}
```